runtime/interpreter/interpreter.go

```diff
@@ -1776,17 +1776,100 @@ func (interpreter *Interpreter) ConvertAndBox(
 	return interpreter.BoxOptional(locationRange, value, targetType)
 }
 
-func (interpreter *Interpreter) convert(value Value, valueType, targetType sema.Type, locationRange LocationRange) Value {
-	if valueType == nil {
-		return value
+func (interpreter *Interpreter) convertStaticType(
+	valueStaticType StaticType,
+	targetSemaType sema.Type,
+) StaticType {
+	switch valueStaticType := valueStaticType.(type) {
+	case ReferenceStaticType:
+		if targetReferenceType, isReferenceType := targetSemaType.(*sema.ReferenceType); isReferenceType {
+			return NewReferenceStaticType(
+				interpreter,
+				ConvertSemaAccesstoStaticAuthorization(interpreter, targetReferenceType.Authorization),
```

Comment:
Is my understanding here correct that `convertStaticType` "converts" a static type to another static type, given a semantic type that must "match" the input static type, and this, the authorization of the reference type, is the only case where the target type's info is used?

Reply:
Could you please add a comment to the function and explain this behaviour a bit more?
That the only different is in using the target reference authorization is very subtle / difficult to notice.